modules/system/rolls/cast-test.js

```diff
@@ -188,11 +188,31 @@ export default class CastTest extends TestWFRP {
       miscastCounter++;
     }
     //@/HOUSE
-
-    this.result.overcasts = Math.max(0, Math.floor(slOver / 2));
+    if (!game.settings.get("wfrp4e", "useWoMOvercast")) {
+      this.result.overcasts = Math.max(0, Math.floor(slOver / 2));
+    } else {
+      this.result.overcasts = Math.max(0, slOver);
+    }
     this.result.overcast.total = this.result.overcasts;
     this.result.overcast.available = this.result.overcasts;
-
+    if (game.settings.get("wfrp4e", "useWoMOvercast")) {
+      if (this.result.overcast.usage.range) {
+        this.result.overcast.usage.range.available = this.result.overcast.available >= game.wfrp4e.config.overCastTable.range[0].cost
+      }
+      if (this.result.overcast.usage.target) {
+        if(this.result.overcast.usage.target.AoE) {
+          this.result.overcast.usage.target.available = this.result.overcast.available >= game.wfrp4e.config.overCastTable.AoE[0].cost
+        } else {
+          this.result.overcast.usage.target.available = this.result.overcast.available >= game.wfrp4e.config.overCastTable.target[0].cost
+        }
+      }
+      if (this.result.overcast.usage.duration) {
+        this.result.overcast.usage.duration.available = this.result.overcast.available >= game.wfrp4e.config.overCastTable.duration[0].cost
+      }
+      if (this.result.overcast.usage.damage) {
+        this.result.overcast.usage.damage.available = this.result.overcast.available >= game.wfrp4e.config.overCastTable.damage[0].cost
+      }
+    }
```

Comment:
This is making me think it would be cleaner to extend the `CastTest` class into a `WoMCastTest` class that could reduce the  if/else hell. Do you think so?

Reply:
Good idea